### PR TITLE
QSPIF Flash Block Device example

### PR DESCRIPTION
**Pull request by offirko**

### CODE REVIEW

**Comment by offirko** [comment]
Hello: @bulislaw , @donatieng , @ARMmbed/mbed-os-maintainers , @dannybenor , @davidsaada .
I'd appreciate your review of the QSPIF Block Device Example

**Comment by 0xc0170** [comment]
Can we rename this repo, as other examples, this should be `mbed-os-example-qspif-blockdevice` or rather qspif as the last word? As we have mbed-os-example-blockdevice, I would assume adding suffix here would identify what blockdevice it is.

I noticed also driver is blockDevice, please fix that as well

**Comment by offirko** [comment]
@0xc0170 - renamed both and made the adjustments, thanks

File: main.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// QSPI SFDP Flash - Block Device example
+#include "mbed.h"
```
Comment (davidsaada) [inline main.cpp:18]: Can we do without this include?

File: main.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+#include "QSPIFBlockDevice.h"
+
+QSPIFBlockDevice block_device(MBED_CONF_QSPIF_QSPI_IO0, MBED_CONF_QSPIF_QSPI_IO1, MBED_CONF_QSPIF_QSPI_IO2, MBED_CONF_QSPIF_QSPI_IO3,
+		MBED_CONF_QSPIF_QSPI_CLK, MBED_CONF_QSPIF_QSPI_CS, QSPIF_POLARITY_MODE_0, MBED_CONF_QSPIF_QSPI_FREQ);
+
+
+int main() {
+    printf("QSPI SFDP Flash Block Device example\n");
```
Comment (davidsaada) [inline main.cpp:26]: Better to have stdio.h included if you use this.

**Review by davidsaada — CHANGES_REQUESTED** [review]
Looks good to me, beside a few petty issue.

File: main.cpp (hunk 1 above)
Comment (offirko) [reply main.cpp:18]: removed

**Pull request merged**